edk2basetools/VfrCompiler/IfrCommon.py

```diff
@@ -0,0 +1,100 @@
+## @file
+# This file is used to define common funtions.
+#
+# Copyright (c) 2022-, Intel Corporation. All rights reserved.<BR>
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+import Common.EdkLogger as EdkLogger
+from VfrCompiler.IfrCtypes import EFI_GUID
+from Common.BuildToolError import PARAMETER_INVALID
+
+# Enumeration of EFI_STATUS.
+RETURN_SUCCESS = EFI_SUCCESS = 0
+EFI_BUFFER_TOO_SMALL = 0x8000000000000000 | (5)
+EFI_ABORTED = 0x8000000000000000 | (21)
+EFI_OUT_OF_RESOURCES = 0x8000000000000000 | (9)
+EFI_INVALID_PARAMETER = 0x8000000000000000 | (2)
+EFI_NOT_FOUND = 0x8000000000000000 | (14)
+RETURN_INVALID_PARAMETER = 0x8000000000000000 | (2)
+RETURN_UNSUPPORTED = 0x8000000000000000 | (3)
+
+ASCII_GUID_BUFFER_INDEX_1 = 8
+ASCII_GUID_BUFFER_INDEX_2 = 13
+ASCII_GUID_BUFFER_INDEX_3 = 18
+ASCII_GUID_BUFFER_INDEX_4 = 23
+ASCII_GUID_BUFFER_MAX_INDEX = 36
+GUID_BUFFER_VALUE_LEN = 11
+
+
+def EFI_ERROR(A):
+    return A < 0
+
+
+# Converts a string to an EFI_GUID.
+def StringToGuid(AsciiGuidBuffer: str, GuidBuffer: EFI_GUID):
```

Comment:
for guid support we have found the uuid module works pretty good and avoids having to write python code that looks like C.  :)
For some examples seem this search.  https://github.com/search?q=repo%3Atianocore%2Fedk2-pytool-library+uuid&type=code